app/repositories/migrations/schema11.ts

```diff
@@ -350,29 +355,62 @@ const migration = (oldRealm: any, newRealm: any) => {
   if (oldRealm.schemaVersion < schemaVersion) {
     const oldInventoryObject = oldRealm.objects('Inventory');
     const newInventoryObject = newRealm.objects('Inventory');
+
     for (const index in oldInventoryObject) {
-      if (oldInventoryObject.inventory_id) {
+      if (
+        oldInventoryObject[index].inventory_id &&
+        (oldInventoryObject[index].status === INCOMPLETE ||
+          oldInventoryObject[index].status === INCOMPLETE_SAMPLE_TREE ||
+          oldInventoryObject[index].status === PENDING_DATA_UPLOAD)
+      ) {
+        // adds all the data from old inventory except APP accessType
         newInventoryObject[index].additionalDetails = oldInventoryObject[
           index
         ].additionalDetails.filter(
           (d: any) => d.accessType === accessTypes.PRIVATE || d.accessType === accessTypes.PUBLIC,
         );
 
-        newInventoryObject[index].appMetadata = JSON.stringify(
-          appAdditionalDataForAPI({ data: oldInventoryObject[index] }),
-        );
+        // adds app version to additional details
+        newInventoryObject[index].additionalDetails.push({
+          key: 'appVersion',
+          value: '1.0.2',
```

Comment:
The next release will have app version 1.0.3 - why is this hardcoded into the migration?

Reply:
All the old registrations should have older version of the app i.e. 1.0.2. But it should be only for status `PENDING_DATA_UPLOAD` not other as that registration was saved in older version. Will add condition for `INCOMPLETE` and `INCOMPLETE_SAMPLE_TREE` to store new version.